sdks/java/core/build.gradle

```diff
@@ -69,7 +69,7 @@ dependencies {
   compile library.java.protobuf_java
   compile library.java.commons_compress
   compile library.java.commons_lang3
-  compile library.java.jsr305
+  shadow library.java.jsr305
```

Comment:
[BeamModulePlugin.groovy#L629](https://github.com/apache/beam/blob/master/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L629) says:
```
    // When the shadowClosure argument is specified, the shadow plugin is enabled to perform shading
    // of commonly found dependencies. Because of this it is important that dependencies are added
    // to the correct configuration. Dependencies should fall into one of these four configurations:
    //  * compile     - Required during compilation or runtime of the main source set.
    //                  This configuration represents all dependencies that much also be shaded away
    //                  otherwise the generated Maven pom will be missing this dependency.
    //  * shadow      - Required during compilation or runtime of the main source set.
    //                  Will become a runtime dependency of the generated Maven pom.
```

JSR305 is needed at runtime ([BEAM-8917](https://issues.apache.org/jira/browse/BEAM-8917)).

Reply:
That comment is incorrect, shadow dependencies will not become runtime dependencies, they will be shaded and not exposed as part of the pom file.

Reply:
Although the word `shadow` is not intuitive, the comment was correct for this sdks/java/core module which has shadowClosure. This PR did fix the problem in snapshot: 
https://repository.apache.org/content/repositories/snapshots/org/apache/beam/beam-sdks-java-core/2.19.0-SNAPSHOT/beam-sdks-java-core-2.19.0-20191218.071919-14.pom
![image](https://user-images.githubusercontent.com/28604/71119854-214b6400-21a9-11ea-93b8-61d52c7bb2b3.png)

javax.annotation.Nullable is not included in beam-sdks-java-core JAR:

```
suztomo@suxtomo24:~/cloud-opensource-java$ jar tf ~/Downloads/beam-sdks-java-core-2.19.0-20191223.072140-19.jar  |grep Nullable
org/apache/beam/sdk/coders/NullableCoder.class
org/apache/beam/sdk/schemas/Schema$EquivalenceNullablePolicy.class
```

With `shadowClosure` specified, only dependencies specified by `shadowClosure > dependencies > include` are shaded. For example protobuf-java is part of shaded dependencies:

```
applyJavaNature(
  automaticModuleName: 'org.apache.beam.sdk',
  shadowClosure: {
    dependencies {
      include(dependency(library.java.protobuf_java))
      include(dependency("org.apache.commons:.*"))
      include(dependency(library.java.antlr_runtime))
    }
```

https://github.com/apache/beam/blob/master/sdks/java/core/build.gradle#L26 .

Reply:
Sorry, I forgot that shadow in beam means unshaded so your totally right. The defaults for the gradle shadow plugin are significantly different.